### PR TITLE
Implement #678 Throw invalid operation if result set is null

### PR DESCRIPTION
**Pull request by joshdrees**

### CODE REVIEW

**Comment by bgrainger** [comment]
Thanks for the contribution! 

I'm not sure why the PR tests are failing; I'll try to run them again.

**Comment by joshdrees** [comment]
Thanks for rerunning them. It looks like the DB tests are still failing to build at the "Testing if container is responding" check, and not running.

**Comment by bgrainger** [comment]
For some reason, the CI script that was working fine has now stopped running on Azure Pipelines. I'm bumping the startup timeout which should address this: e56a939d0bf0ef85043a3ddf9faa4f0db3255818.

File: src/MySqlConnector/MySql.Data.MySqlClient/MySqlDataReader.cs [hunk 1]
```diff
@@ -273,6 +273,8 @@ public override long GetChars(int ordinal, long dataOffset, char[] buffer, int b
 #if !NETSTANDARD1_3
 		public override DataTable GetSchemaTable()
 		{
+			if (m_resultSet is null)
+				throw new InvalidOperationException("There is no current result set.");
```
Comment (bgrainger) [inline src/MySqlConnector/MySql.Data.MySqlClient/MySqlDataReader.cs]: `BuildSchemaTable` calls `GetResultSet`, which already performs this check.

I don't know offhand where the test needs to happen, but this isn't the right test.

File: tests/Conformance.Tests/DataReaderTests.cs [hunk 2]
```diff
@@ -12,8 +12,5 @@ public DataReaderTests(SelectValueFixture fixture)
 
 		[Fact(Skip = "Deliberately throws InvalidCastException")]
 		public override void GetTextReader_returns_empty_for_null_String() { }
-
-		[Fact(Skip = "https://github.com/mysql-net/MySqlConnector/issues/678")]
-		public override void GetSchemaTable_throws_after_Delete() { }
```
Comment (bgrainger) [inline tests/Conformance.Tests/DataReaderTests.cs]: I added this test to the PR, which tests the completed functionality. (It's currently failing.)

**Comment by bgrainger** [comment]
The remaining build failure is a permissions problem (in Azure Pipelines) for all PRs.

**Pull request merged**